Userland/Libraries/LibGfx/Rect.h

```diff
@@ -557,6 +557,24 @@ class Rect {
         return points;
     }
 
+    template<typename U = T>
+    [[nodiscard]] Gfx::Rect<U> interpolated_to(Gfx::Rect<T> const& to, float factor) const
+    {
+        VERIFY(factor >= 0.0f);
+        VERIFY(factor <= 1.0f);
+        if (factor == 0.0f)
+            return *this;
+        if (factor == 1.0f)
+            return to;
+        if (this == &to)
+            return *this;
+        auto interpolated_left = round_to<U>(x() + ((to.x() - x()) * factor));
```

Comment:
use AK::mix

Reply:
Ah nice, didn't know we had this helper.